chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -636,6 +636,86 @@ Array Swapaxes(const Array& a, int8_t axis1, int8_t axis2) {
     return out;
 }
 
+Array Repeat(const Array& a, int64_t repeats, nonstd::optional<int8_t> axis) {
+    if (repeats < 0) {
+        throw DimensionError("repeats must be larger than 0.");
+    }
+
+    int8_t targetAxis = 0;
+    Array targetArray;
+
+    if (axis.has_value()) {
+        if (*axis < -static_cast<int32_t>(a.shape().size())) {
+            throw DimensionError("axis must be larger than - the number of shape.");
+        }
+
+        if (*axis >= static_cast<int32_t>(a.shape().size())) {
+            throw DimensionError("axis must be lower than the number of shape.");
+        }
+
+        targetAxis = *axis;
+
+        if (targetAxis < 0) targetAxis = static_cast<int32_t>(a.shape().size()) + targetAxis;
```

Comment:
We have a `NormalizeAxis` that does all of this :)

Reply:
fixed